main.py

```diff
@@ -0,0 +1,364 @@
+import requests
+import json
+import datetime
+import os
+from operator import itemgetter
+
+
+# GLOBALS
+PATH = os.path.dirname(os.path.realpath(__file__))   # Gets the absolute path for the running file
+RAW_PATH = PATH + "/data/{}/raw/"
+EDIT_PATH = PATH + "/data/{}/edited/"
+FILE_NAME = "Riven_data_{}_{}.json"
+TOTAL_FILE_PATH = EDIT_PATH + "TOTAL_{}.json"
+L_MONDAY = ""
+PLATFORMS = ["PC", "PS4", "XB1", "SWI"]
+URL = "https://n9e5v4d8.ssl.hwcdn.net/repos/weeklyRivens{}.json"
+
+
+def last_monday():
+    '''Gets the last monday date'''
+    today = datetime.date.today()
+    monday = today - datetime.timedelta(days=today.weekday())
+    return monday
+
+
+def get_riven_json(platform):
+    '''Gets the json for a given platform'''
+    print(f"Checking if this week json for {platform} is saved!")
+    filename = FILE_NAME.format(platform, L_MONDAY.strftime("%y-%m-%d"))
+    file_path = f"{RAW_PATH.format(platform)}{filename}"
+
+    # Checks if the file already exists
+    if not os.path.isfile(file_path):
+        print("No file found, creating it!\n-----------------")
+        r = requests.get(URL.format(platform))
+        r.raise_for_status()
+        data = r.json()
+        with open(file_path, "w") as outfile:
+            json.dump(data, outfile, indent=4)
+    else:
```

Comment:
This is a bit of stylistic choice, so its not for everyone, just sharing my two cents. I was taught to have early returns in JS, as to not make super deep nested blocks (a *feature* of JS 😆 ).

E.g.:

```py
if os.path.isfile(file_path):
  print('already exist!')
  return

# continue logic, as if file_path does not exist
print('no file, creating...')
```

Reply:
So the whole part would look like this?
```suggestion
    if os.path.isfile(file_path):
        print("File already exists!\n-----------------")
        return

    print("No file found, creating it!\n-----------------")
    r = requests.get(URL.format(platform))
    r.raise_for_status()
    data = r.json()
    with open(file_path, "w") as outfile:
        json.dump(data, outfile, indent=4)
```

Reply:
Aye exactly 😄